app/utils.py

```diff
@@ -91,250 +109,335 @@ def fetch_and_store_players():
+                parsed_age = age_parser(age)
+                Player.add_player(
+                    player_id=int(player_id),
+                    name=player_name,
+                    position=position,
+                    weight=weight,
+                    born_date=born_date,
+                    age=parsed_age,
+                    exp=exp,
+                    school=school,
+                    available_seasons="Unknown",
+                )
+
+                # Add player to the team's roster in the database
+                team_obj.add_to_roster(
+                    player_id=player_id,
+                    player_name=player_name,
+                    player_number=player_number,
+                    position=position,
+                    how_acquired=how_acquired,
+                    season=season,
+                )
+
+        except Exception as e:
+            logger.error(
+                "Error fetching roster for team %s (ID: %s): %s", team_name, team_id, e
+            )
+
+    logger.info(
+        """All current NBA teams and their rosters have been
+                successfully stored."""
+    )
+
+
+def age_parser(age):
+    """
+    Parses the given age value and returns it as an integer if valid.
+
+    This function checks if the input `age` is a string containing
+    only digits. If so, it converts the string to an integer and
+    returns it. Otherwise, it returns `None`.
+
+    Args:
+        age (str | int): The age value to parse. Typically expected to be
+            a string representing a number.
+
+    Returns:
+        int | None: The parsed age as an integer if valid, or `None` if
+            the input is not a valid age string.
+    """
+    parsed_age = int(age) if isinstance(age, str) and age.isdigit() else None
+    return parsed_age
 
 
 def fetch_and_store_all_players_stats():
     """Fetch stats for all active players."""
     all_players = players.get_active_players()
+    logger.info("Fetched {len(all_players)} active players from NBA API.")
+
     for player in all_players:
         fetch_and_store_player_stats(player["id"])
-    print("All active player statistics have been successfully stored.")
+    logger.info("All active player statistics have been successfully stored.")
+
 
 def update_player_teams():
-    """Update the team field in the players table using data from the roster table."""
-    cur.execute('''
-        UPDATE players
-        SET team = teams.name
-        FROM roster
-        INNER JOIN teams ON roster.team_id = teams.team_id
-        WHERE players.player_id = roster.player_id;
-    ''')
-    conn.commit()
-    print("All currently rostered players have been listed for their correct team.")
+    """
+    Update the team field in the players table using data from the roster
+    table.
+    """
+    try:
+        update_query = """
+            UPDATE players
+            SET team = teams.name
```

Comment:
Not a good way to store TEAMS, we have a rosters table for a reason. what if a player is on the cavs then the lakers the next year??? we have to keep team separate from player table for this reason

Reply:
Deleted method